### PR TITLE
quazip: revert to 0.9.1 & add quazip1 package for v1.1

### PR DESCRIPTION
**Pull request by tenzap**
The way this is done may still need to be discussed in the ticket

* This reverts commit b783795d8de942ec76149db5a62c550c50f409cd.
* increment epoch to quazip package
* add quazip1 package for version 1.1

Closes: https://trac.macports.org/ticket/63939

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.11.6 15G22010 x86_64
Xcode 7.3.1 7D1014

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by reneeotten** [comment]
if both dependents in MacPorts (i.e., `KeePassXC` and `texstudio`) build fine with the new version I personally don't see an obvious reason to keep the old one and create a new `quazip1` port. Or am I missing something here?

**Comment by tenzap** [comment]
KeePassXC 2.6.6 doesn't know how to use quazip >= 1.0 and would need patching so that it can compile with it. I prepared such a patch, but as said [here also mentioned in the ticket 63939](​https://github.com/stachenov/quazip/blob/master/QuaZip-1.x-migration.md), it seems each package using quazip has to proceed with some changes anyway to upgrade from pre1.0 to >=1.0

Concerning texstudio I don't know.

**Comment by reneeotten** [comment]
it seems there is a patch already for KeepassXC: https://github.com/keepassxreboot/keepassxc/issues/5534

**Comment by tenzap** [comment]
The thing is it doesn't work with 2.6.6 on macports

```
-- Could NOT find QuaZip (missing: QUAZIP_INCLUDE_DIRS) 
-- KeeShare: Secure container support is DISABLED; quazip library not found
```
log: https://build.macports.org/builders/ports-10.13_x86_64-builder/builds/129260/steps/install-port/logs/stdio/text

**Comment by reneeotten** [comment]
I am confused: did you add that patch already, and how about specifying the include directory as configure argument?

**Comment by tenzap** [comment]
This ticket is about quazip. The developer says that 1.x should be an update of 0.9x and that both versions should be installed in parallel. That's what I'm trying to achieved trough this PR. Originally I create ticket https://trac.macports.org/ticket/63939 to discuss it.

Concerning KeePassXC, v2.6.6 as provided by upstream doesn't detect quazip (whether 0.9x or 1.x) on macports.
For this I prepared a patch for which there is no PR yet. To add it, I prefer to wait until it is decided how quazip will be packaged in macports. Then I can also set the quazip package to depend on at the same time.

**Comment by reneeotten** [comment]
> This ticket is about quazip. The developer says that 1.x _should not_ be an update of 0.9x and that both versions should be installed in parallel. That's what I'm trying to achieved trough this PR. Originally I created ticket https://trac.macports.org/ticket/63939 to discuss it.

Yes, I understand that and sure one could package both version in parallel if there is a need to do so. My point is that I don't see that need if the only two dependents of this port build fine with this new version.

**Comment by tenzap** [comment]
@reneeotten , @herbygillot could you have a look at the new comments? Can we merge this?

**Comment by mascguy** [comment]
> @reneeotten , @herbygillot could you have a look at the new comments? Can we merge this?

Folks, any final thoughts/concerns at this point?

**Review by mascguy — APPROVED** [review]

**Comment by reneeotten** [comment]
> > @reneeotten , @herbygillot could you have a look at the new comments? Can we merge this?
> 
> Folks, any final thoughts/concerns at this point?

I think my opinion was clear enough and I suggested how do this *if* we decided to do this. None if that was done so I’d appreciate if you would not merge PRs like this.

**Comment by mascguy** [comment]
> I think my opinion was clear enough and I suggested how do this _if_ we decided to do this. None if that was done so I’d appreciate if you would not merge PRs like this.

For cases where there are concerns related to the approach for an open PR - beyond just unresolved comments/changes (which we check for before merging) - should we also add the "DO NOT MERGE YET" label? That would make it crystal-clear that a given PR, in it's current state, shouldn't be merged.

Thoughts?

**Comment by mascguy** [comment]
p.s. I'm trying to balance the wishes of members and reviewers, while also advocating for contributors. (And trying to assist contributors, to help avoid situations where their PRs remain idle for days and weeks... without any clear consensus, or path forward.) And it's a struggle at times.

Thoughts/ideas as to how we can improve our process?

**Comment by mascguy** [comment]
As for the specifics related to this PR: There was a difference of opinion on the approach, which boiled down to whether we maintain two versions of `quazip`.

Based on the points brought up by @tenzap, the details in the associated ticket, etc, maintaining two versions of the port makes sense to me. And while it's certainly preferable to avoid multiple port versions when possible/feasible, it doesn't sound like it in this particular case.

If the two ports conflicted - which causes endless frustration for our users - I would have pushed back. But given that both can be installed side-by-side - and exist harmoniously - that eliminated my concerns.

Renee, I have a tremendous amount of respect and appreciation for everything you bring to MacPorts. And I tend to defer to your opinions 99+% of the time, as you're usually right! This is one of the rare cases where I'll have to respectfully disagree with you though.  ;-)

**Comment by reneeotten** [comment]
if there is a need for both ports that is fine as I mentioned; we should follow then the usual way of naming them having `quazip` follow the latest upstream version as I mentioned above. 

The process we typically follow is that concern/questions in an PR from a reviewer get answered/worked out and then it can be merged by anyone. If you ask the opinion of @herbygillot and me and then decide to merge anyway within an hour without waiting for a response, that's not the way to do things.

**Comment by mascguy** [comment]
> if there is a need for both ports that is fine as I mentioned; we should follow then the usual way of naming them having `quazip` follow the latest upstream version as I mentioned above.

If that's the key outstanding concern, then we can assist @tenzap with addressing that.

I originally thought that you were opposed to two versions overall. Glad to hear!

**Comment by reneeotten** [comment]
well, right now we can as well leave it like this as the “epoch” has been added already anyway. Let’s decide upon the next update or so…

the other concern here and at some other places is regarding asking a question and then merging anyway without waiting for an answer. People have day-jobs and Thanksgiving shopping to do as well 😉. Other than that, I do appreciate your efforts in keeping the PR queue moving!

**Pull request merged**
